File: dev-docs/snippets/auth/enable-offline-access.swift
import JsBaoClient

// Enable offline access via the typed `client.auth` namespace. The result is
// a typed `EnableOfflineAccessResult` (`enabled`, `method?`, `reason?`).
// `EnableOfflineAccessOptions` accepts `preferBiometric` / `allowPinFallback` /
// `ttlDays` / `retention` / `pinProvider`, matching JS.
func enableOfflineAccess(client: JsBaoClient) async throws {
  // #region example
  let result = try await client.auth.enableOfflineAccess(
    options: EnableOfflineAccessOptions(
      preferBiometric: true,
      allowPinFallback: true,
      ttlDays: 14
    )
  )
  let enabled = result.enabled
  let method = result.method
  // #endregion example
  _ = (enabled, method)
}

File: dev-docs/snippets/auth/enable-offline-access.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Enable offline access using a passkey (largeBlob) or PIN-based grant.
export async function enableOfflineAccess(client: JsBaoClient) {
  // #region example
  const { enabled, method } = await client.enableOfflineAccess({
    preferBiometric: true,
    allowPinFallback: true,
    ttlDays: 14,
  });
  // #endregion example
  return { enabled, method };
}
